src/fiatExchanges/utils.tsx

```diff
@@ -291,3 +294,78 @@ export const filterProvidersByPaymentMethod = (
 ) => {
   return externalProviders?.find((quote) => quote.paymentMethods.includes(paymentMethod))
 }
+
+export interface ProviderSelectionAnalyticsData {
+  centralizedExchangesAvailable: boolean
+  totalOptions: number
+  paymentMethodsAvailable: Record<PaymentMethod, boolean>
+  transferCryptoAmount: string
+  lowestFeeKycRequired: boolean | undefined
+  lowestFeeCryptoAmount: string | undefined
+  lowestFeeProvider: string | undefined
+  lowestFeePaymentMethod: PaymentMethod | undefined
+}
+
+/**
+ * Get analytics data for provider selection.
+ *
+ * Used for cico_providers_quote_selected and cico_providers_exchanges_selected analytics events.
+ *
+ * @param normalizedQuotes
+ * @param exchangeRates
+ * @param legacyMobileMoneyProviders
+ * @param centralizedExchanges
+ * @param transferCryptoAmount
+ */
+export function getProviderSelectionAnalyticsData({
+  normalizedQuotes,
+  exchangeRates,
+  legacyMobileMoneyProviders,
+  centralizedExchanges,
+  transferCryptoAmount,
+}: {
+  normalizedQuotes: NormalizedQuote[]
+  exchangeRates: { [token in Currency]: string | null }
+  legacyMobileMoneyProviders?: LegacyMobileMoneyProvider[]
+  centralizedExchanges?: ExternalExchangeProvider[]
+  transferCryptoAmount: string
+}): ProviderSelectionAnalyticsData {
+  let lowestFeePaymentMethod: PaymentMethod | undefined = undefined
+  let lowestFeeProvider: string | undefined = undefined
+  let lowestFeeCryptoAmount: BigNumber | null = null
+  let lowestFeeKycRequired: boolean | undefined = undefined
+  const centralizedExchangesAvailable = !!centralizedExchanges && centralizedExchanges?.length > 0
+  const paymentMethodsAvailable: Record<PaymentMethod, boolean> = {
+    [PaymentMethod.Bank]: false,
+    [PaymentMethod.Card]: false,
+    [PaymentMethod.Coinbase]: false,
```

Comment:
It doesn't look like coinbase can ever be true

Reply:
coinbase is treated as an "external provider" in the SelectProviders page, so it should be in the list of normalized quotes when it is offered.

Reply:
Unfortunately it is not because of these two lines of code

https://github.com/valora-inc/wallet/blob/main/src/fiatExchanges/quotes/normalizeQuotes.ts#L87

https://github.com/valora-inc/cloud-functions/blob/main/src/cico/fetchProviders.ts#L209

Reply:
There's no quote to normalize for CoinbasePay

Reply:
Oh ok, thanks for spotting that. I'll work on a fix and a regression for that issue.